lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+# WDL Lint Rules
+
+The below rules will all be added as lints to `wdl-grammar` and `wdl-ast` (if they aren't already). Most rules will be assigned a "group" that they belong to. The intention of having these groups is twofold. One, it will allow an easy method for disabling multiple related rules at the same time. Two, it will make the documentation easier to organize and search. We're still refining what these final groups will be, so not every rule in this document will be assigned one yet. Some rules may never be assigned as a group if they are truly stand-alone. The current groups are: `spacing`, `naming`, `sorting`, `completeness`, `container`, and `deprecated`.
+
+This document only concerns what we are calling "lint warnings". Lint warnings are distinct from what we call "validation errors". A validation error results when the specification (v1.x at time of writing) is not followed. These errors render your WDL document invalid. On the other hand, lint warnings are just our (the writers of `wdl-grammar` and `wdl-ast`) opinion on what "good" WDL should look like. They are matters of style, readability, consistency, or for encouraging proper documentation. Your document may be littered with lint warnings, but if there aren't any validation errors, then your document is valid WDL.
+
+We are aiming to automate as much as we can to make adherence to these lint rules as easy as possible. An "auto-formatter" will be packaged as part of our [`sprocket` tool](https://github.com/stjude-rust-labs/sprocket). Auto-formatting has not yet been implemented as this project is still early in development, but while reading this document, please assume that anything easily automated (and perhaps complex to do manually) will eventually be handled by the formatter built into `sprocket`.
+
+Some of these lint warnings are arbitrary. There are times where several different methods for styling a document could be considered equally appealing. The important part is picking one such style and adhering to it consistenly across a document or codebase. In such cases, this document will only present one option. In the future, that singular option may just be the default, with several more-or-less equivalent styles presented as alternative options available through configuration. If you feel we are picking the wrong defaults, please let us know what you think would be better!
+
+At the time of writing (January 2024), we consider all rules herein to be up for debate. Please open an issue on this repository if you would like to suggest changes to this document.
+
+## Rules
+
+### `version_declaration_placement`
```

Comment:
The `version` line should be followed by *exactly* one blank line (this rule should enforce this rather than the `import` rules that follow since there might not be any imports)